emle/models.py

```diff
@@ -1210,3 +1223,394 @@ def forward(self, atomic_numbers, charges_mm, xyz_qm, xyz_mm):
+            self.register_buffer("node_attrs", node_attrs.to(self._dtype))
+            self.register_buffer("batch",_torch.zeros(node_attrs.shape[0], dtype=_torch.long, requires_grad=False),)
+
+            print("Node attributes:", node_attrs.shape)
+            print("Batch:", self.batch.shape)
+            print("Ptr:", self.ptr.shape)
+            print(node_attrs)
+
+        # No PBCs for now.
+        self.register_buffer("pbc",_torch.tensor([False, False, False], dtype=_torch.bool, requires_grad=False),)
+        self.register_buffer("cell", _torch.zeros((3, 3), dtype=self._dtype, requires_grad=False))
+
+        # Assign a tensor attribute that can be used for assigning the AEVs.
+        self._aev_computer._aev = _torch.empty(0, device=device)
+    
+    @staticmethod
+    def _to_one_hot(indices: _torch.Tensor, num_classes: int) -> _torch.Tensor:
+        """
+        Convert a tensor of indices to one-hot encoding.
+
+        Parameters
+        ----------
+        indices: torch.Tensor
+            Tensor of indices.
+        num_classes: int
+            Number of classes of atomic numbers.
+        
+        Returns
+        -------
+        oh: torch.Tensor
+            One-hot encoding of the indices.
+        """
+        shape = indices.shape[:-1] + (num_classes,)
+        oh = _torch.zeros(shape, device=indices.device).view(shape)
+        return oh.scatter_(dim=-1, index=indices, value=1)
+
+    @staticmethod
+    def _atomic_numbers_to_indices(atomic_numbers: _torch.Tensor, z_table: List[int]) -> _torch.Tensor:
+        """
+        Get the indices of the atomic numbers in the z_table.
+
+        Parameters
+        ----------
+        atomic_numbers: torch.Tensor (N_ATOMS,)
+            Atomic numbers of QM atoms.
+        z_table: List[int]
+            List of atomic numbers in the MACE model.
+
+        Returns
+        -------
+        indices: torch.Tensor (N_ATOMS, 1)
+            Indices of the atomic numbers in the z_table.
+        """
+        return _torch.tensor(
+            [z_table.index(z) for z in atomic_numbers], dtype=_torch.long
+        ).unsqueeze(-1)
+
+    def _get_node_attrs(self, atomic_numbers: _torch.Tensor) -> _torch.Tensor:
+        """
+        Internal method to get the node attributes for the MACE model.
+
+        Parameters
+        ----------
+        atomic_numbers: torch.Tensor (N_ATOMS,)
+            Atomic numbers of QM atoms.
+
+        Returns
+        -------
+        node_attrs: torch.Tensor (N_ATOMS, N_FEATURES)
+            Node attributes for the MACE model.
+        """        
+        ids = self._atomic_numbers_to_indices(atomic_numbers, z_table=self._z_table)
+        return self._to_one_hot(ids, num_classes=len(self._z_table))
+    
+    def _get_neighbor_pairs(self, positions: _torch.Tensor, cell: Optional[_torch.Tensor]) -> Tuple[_torch.Tensor, _torch.Tensor]:
```

Comment:
It looks like this could be made static by adding `dtype`, `device`, and the cutoff as kwargs, which might make it usable in other contexts. If you think it could be more generally useful, then it could be moved to a method of the `EMLE` base class.

Reply:
It can certainly be useful in other contexts. I've made it static but will leave it in the `MACEEMLE` class for now. It can be easily moved to the base class if needed in the future.

Reply:
Great. I'll think about moving it when I refactor.